sample-code/javascript-wd/test/basic/android-basic-interactions.test.js

```diff
@@ -0,0 +1,63 @@
+import wd from 'wd';
+import chai from 'chai';
+import { androidCaps, serverConfig } from '../helpers/caps';
+
+const {assert} = chai;
+
+const PACKAGE = 'io.appium.android.apis';
+const SEARCH_ACTIVITY = '.app.SearchInvoke';
+const ALERT_DIALOG_ACTIVITY = '.app.AlertDialogSamples';
+
+describe('Basic Android interactions', function () {
+
+  let driver;
+
+  before(async function () {
+    // Connect to Appium server
+    driver = await wd.promiseChainRemote(serverConfig);
+
+    // Start the session
+    await driver.init({
+      ...androidCaps,
+      app: require('../helpers/apps').androidApiDemos,
+      appActivity: SEARCH_ACTIVITY, // Android-specific capability. Can open a specific activity.
+    });
+  });
+
+  after(async function () {
+    await driver.quit();
+  });
+
+  it('should send keys to search box and then check the value', async function () {
+    // Enter text in a search box
+    const searchBoxElement = await driver.elementById('txt_query_prefill');
+    await searchBoxElement.sendKeys('Hello world!');
+
+    // Press on 'onSearchRequestedButton'
+    const onSearchRequestedButton = await driver.elementById('btn_start_search');
+    await onSearchRequestedButton.click();
+
+    // Check that the text matches the search term
+    await driver.waitForElementById('android:id/search_src_text');
+    const searchText = await driver.elementById('android:id/search_src_text');
```

Comment:
Can these two lines not just be 
```js
const searchText = await driver.waitForElementById('android:id/search_src_text');
```

Reply:
Aha, I didn't realize that `waitFor` actually returned the element.

Reply:
I _think_ it does. The code seems to indicate so, but it's not the easiest code to read. I would give it a try.